### DISCUSSION TITLE
Monologue

### DISCUSSION THREAD

**Discussion by guiaugustinegri** [souzatharsis/podcastfy#152] [Q&A] [answered]
is there any way to set it up to be a monologue?

**Comment by souzatharsis** [comment]
Interesting....

Yes?!?

Why don't you try the configuration below.
Here, I am setting both speakers to use the same openai voice.
Additionally, I am providing some custom instructions telling the LLM this
is a monologue.

Let me know what you think; curious about the result....

monolog_config = {
    "conversation_style": ["monolog"],
    "roles_person1": "monolog speaker1",
    "roles_person2": "same as person1: monolog speaker1",
    "creativity": 1.0

        "text_to_speech": {
            "openai": {
                "default_voices": {
                    "question": "echo",
                    "answer": "echo"
                }
            }
        }

    "user_instructions": "The transcript should be a long form speech
delivered by a single character . Hence both Person1 and Person2
should be the same character. Speaker should be talking with himself!"




<http://linkedin.com/in/tharsissouza>

**Reply by guiaugustinegri** [reply]
Thanks for the answer, and congratulations on the app, fantastic work =)

I tested it, it kept the same voice, but in the model I used it ends up changing the style a little, which creates a dissonance when listening. The flow doesn't seem like a monologue, because in the conversation there is a cue for another person to speak and mention the other participant by name.

I tested it and created a specific user_instructions so as not to mention by name, which helped, but it still doesn't seem like a monologue flow.

https://audio.com/guiherme-negri/audio/hor2

Somehow it "forces" it to go into this format of a conversation between two people. Is there anything I can do to remove this force?

Ty

------

The code used:

from podcastfy.client import generate_podcast

# Texto para o monólogo
texto = """
Imagine um mundo onde desafios complexos não são solucionados por um punhado de mentes brilhantes, mas por uma coletivo de inteligências, cada uma contribuindo com sua especialidade única para criar soluções. Este não é um cenário tirado de uma obra de ficção científica, mas a promessa iminente dos enxames de agentes de IA para o nosso futuro próximo.

À medida que refletimos sobre a velocidade vertiginosa da inovação em IA, fica claro que estamos no limiar de uma revolução não apenas tecnológica, mas também existencial. Estamos começando a criar agentes de IA, pequenas entidades autônomas, que trabalham em conjunto em verdadeiros "enxames". Esses enxames não são simples agrupamentos de tecnologias; são ecossistemas inteligentes capazes de aprender, adaptar-se e evoluir.

O Poder dos Enxames
Os enxames de agentes são como cardumes ou revoadas de pequenas inteligências artificiais que, livres de intervenção humana direta, são capazes de executar tarefas específicas com uma precisão surpreendente. Eles aprendem uns com os outros em tempo real, trocam informações e ajustam suas estratégias. Esta colaboração resulta em uma eficiência que transcende a soma de suas partes individuais, um verdadeiro exemplo de propriedades emergentes em ação. A beleza desse sistema é que ele é desprovido de ego, focado unicamente na otimização e na resolução de problemas.

Transformando Setores
Utilizando o exemplo do campo da medicina: diariamente são publicados milhares de novos estudos científicos. Um único médico acharia impossível acompanhar tal volume de informação. Aqui, os enxames de agentes entram em cena, cada um com uma função específica, desde compilar pesquisas recentes até analisar tratamentos existentes. Eles operam incessantemente, permitindo avanços médicos antes inimagináveis.

A Era da Criação
As possibilidades são tão vastas quanto nossa imaginação. Sundar Pichai, CEO do Google, comparou recentemente o potencial da IA generativa à invenção do fogo. Não porque trará destruição, mas porque tem o potencial de iluminar, aquecer e transformar fundamentalmente nossa sociedade. Veja que ele não comparou com a invenção da máquina a vapor, dos carros, rádio, televisão ou internet. Ele comparou a importância para a humanidade ao descobrimento do fogo.
"""

# Configurações personalizadas para usar ElevenLabs
monolog_config = {
    "conversation_style": ["monolog"],
    "roles_person1": "Horacio",  # Usando a voz do Horacio
    "roles_person2": "same as person1: Horacio",   # Mesma voz para manter o monólogo
    "creativity": 1.0,
    "text_to_speech": {
        "elevenlabs": {  # Mudando para ElevenLabs
            "default_voices": {
                "question": "Horacio",  # Voz do apresentador
                "answer": "Horacio"      # Usando a mesma voz para manter o monólogo
            },
            "model": "eleven_multilingual_v2"  # Modelo multilíngue do ElevenLabs
        }
    },
    "output_language": "Portuguese",  # Configurando o idioma de saída
    "user_instructions": "Este é um monólogo onde um único personagem fala, sem mencionar nomes de terceiros ou mesmo o próprio nome. O foco deve estar exclusivamente no conteúdo do monólogo, sem passagens entre diferentes vozes ou personagens."
}

# Geração do podcast
generate_podcast(
    text=texto,  # Passando o texto diretamente
    conversation_config=monolog_config,
    tts_model="elevenlabs"  # Especificando o modelo de TTS
)

**Comment by souzatharsis** [comment, marked as answer]
Yes. Podcastfy was designed for a multispeaker podcast generation use case.
Here, we are making a gambiarra to turn it into a monologue.

:)

There is an open issue to generalize the number of speakers. That's an
interesting problem to solve.

<http://linkedin.com/in/tharsissouza>
Thanks for the discussion!

**Reply by guiaugustinegri** [reply]
Justo, thanks for the answer.
It will be really cool in the future to set the number of participants and have the possibility of a monologue.

Until then, we'll use the gambiarra ahah =p
Thanks
